apps/dashboard/app/javascript/packs/files/data_table.js

```diff
@@ -0,0 +1,441 @@
+import 'datatables.net';
+import 'datatables.net-bs4/js/dataTables.bootstrap4';
+import 'datatables.net-select';
+import 'datatables.net-select-bs4';
+import Handlebars from 'handlebars';
+import {EVENTNAME as CLIPBOARD_EVENTNAME} from './clip_board.js';
+import {EVENTNAME as FILEOPS_EVENTNAME} from './file_ops.js';
+import {EVENTNAME as SWAL_EVENTNAME} from './sweet_alert.js';
+
+export { CONTENTID, EVENTNAME };
+
+const EVENTNAME = {
+    getJsonResponse: 'getJsonResponse',
+    reloadTable: 'reloadTable',
+    goto: 'goto',
+};
```

Comment:
I know we had a conversation on event names and owners, but I'm now realizing that events are bi directional and that's what was tripping us up.

Maybe it's more like a file like data_table exports two things, one for the events it listens to, what you have, and the events it itself publishes.
```
// events this object publishes
const PUBLISHING_EVENTS = {}

// events this object listens too. 
const LISTENING_EVENTS = {}
```

Reply:
It does seem cleaner and easier to manage.  Good call.

Reply:
After looking at the code again. I already have the events broken down to class, so there is really no need to break it down any further.  We are dealing with no more than 5 event names, so to break down further in this scenario, i do not believe is necessary.  However, if we get to a point where that is needed, then I'll address it then.